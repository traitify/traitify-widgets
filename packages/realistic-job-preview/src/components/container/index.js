import PropTypes from "prop-types";
import Context from "lib/context";
import ErrorBoundary from "./error-boundary";
import Theme from "./theme";
import useReducer from "./use-reducer";

function Container({children, className = null, ...props}) {
  const state = useReducer({...props});

  return (
    <Context.Provider value={state}>
      <Theme className={className}>
        <ErrorBoundary>{children}</ErrorBoundary>
      </Theme>
    </Context.Provider>
  );
}

Container.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string
};

export default Container;
