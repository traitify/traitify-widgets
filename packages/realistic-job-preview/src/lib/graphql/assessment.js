export const create = `
  mutation createAssessment(
    $localeKey: String!,
    $profileID: String!,
    $surveyID: String!
  ) {
    createAssessment(
      localeKey: $localeKey,
      profileId: $profileID,
      surveyId: $surveyID
    ) {
      completedAt
      failedReason
      id
      insertedAt
      instructions
      isFit
      jobId
      localeKey
      profileId
      responses {
        id
        questionId
        questionText
        responseOptions {
          responseOptionId
          responseOptionText
        }
        selectedResponseOptionId
        selectedResponseOptionText
      }
      startedAt
      status
      surveyId
      totalCorrectResponses
      updatedAt
      videos {
        thumbnailUrl
        videoUrl
      }
    }
  }
`;

export const get = `
  query assessment(
    $assessmentID: ID!
  ) {
    assessment(
      id: $assessmentID
    ) {
      completedAt
      failedReason
      id
      insertedAt
      instructions
      isFit
      jobId
      localeKey
      profileId
      responses {
        id
        questionId
        questionText
        responseOptions {
          responseOptionId
          responseOptionText
        }
        selectedResponseOptionId
        selectedResponseOptionText
      }
      startedAt
      status
      surveyId
      totalCorrectResponses
      updatedAt
      videos {
        thumbnailUrl
        videoUrl
      }
    }
  }
`;

export const path = "/realistic-job-previews/graphql";

export const start = `
  mutation startAssessment(
    $assessmentID: ID!
  ) {
    startAssessment(
      id: $assessmentID
    ) {
      completedAt
      failedReason
      id
      insertedAt
      instructions
      isFit
      jobId
      localeKey
      profileId
      responses {
        id
        questionId
        questionText
        responseOptions {
          responseOptionId
          responseOptionText
        }
        selectedResponseOptionId
        selectedResponseOptionText
      }
      startedAt
      status
      surveyId
      totalCorrectResponses
      updatedAt
      videos {
        thumbnailUrl
        videoUrl
      }
    }
  }
`;

export const update = `
  mutation updateAssessmentAnswer(
    $answers: [AssessmentAnswer],
    $assessmentID: String!
  ) {
    updateAssessmentAnswer(
      answers: $answers,
      assessmentId: $assessmentID
    ) {
      completedAt
      failedReason
      id
      insertedAt
      instructions
      isFit
      jobId
      localeKey
      profileId
      responses {
        id
        questionId
        questionText
        responseOptions {
          responseOptionId
          responseOptionText
        }
        selectedResponseOptionId
        selectedResponseOptionText
      }
      startedAt
      status
      surveyId
      totalCorrectResponses
      updatedAt
      videos {
        thumbnailUrl
        videoUrl
      }
    }
  }
`;
