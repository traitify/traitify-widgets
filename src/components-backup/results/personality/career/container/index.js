import PropTypes from "prop-types";
import TraitifyPropTypes from "lib/helpers/prop-types";
import useDidMount from "lib/hooks/use-did-mount";
import useDidUpdate from "lib/hooks/use-did-update";
import withTraitify from "lib/with-traitify";
import CareerFilter from "../filter";
import CareerList from "../list";
import CareerModal from "../modal";
import style from "./style.scss";

function CareerContainer(props) {
  const {ui, isReady, setElement} = props;
  const state = {};

  useDidMount(() => { ui.trigger("Careers.initialized", {props, state}); });
  useDidUpdate(() => { ui.trigger("Careers.updated", {props, state}); });

  return (
    isReady("results") && (
      <div className={style.container} ref={setElement}>
        <CareerFilter {...props} />
        <CareerList {...props} />
        <CareerModal {...props} />
      </div>
    )
  );
}
CareerContainer.propTypes = {
  isReady: PropTypes.func.isRequired,
  ui: TraitifyPropTypes.ui.isRequired,
  setElement: PropTypes.func.isRequired
};

export {CareerContainer as Component};
export default withTraitify(CareerContainer);
