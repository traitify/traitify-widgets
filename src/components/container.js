/* eslint-disable react/forbid-prop-types */
import PropTypes from "prop-types";
import {useEffect} from "react";
import {RecoilRoot, useRecoilState, useSetRecoilState} from "recoil";
import slice from "lib/common/object/slice";
import Http from "lib/http";
import I18n from "lib/i18n";
import Listener from "lib/listener";
import {
  assessmentIDState,
  benchmarkIDState,
  httpState,
  i18nState,
  listenerState,
  loadingState,
  localeState,
  optionsState,
  profileIDState
} from "lib/recoil";

function State({children, http: _http, i18n: _i18n, listener: _listener, options}) {
  const setAssessmentID = useSetRecoilState(assessmentIDState);
  const setBenchmarkID = useSetRecoilState(benchmarkIDState);
  const setHttp = useSetRecoilState(httpState);
  const setI18n = useSetRecoilState(i18nState);
  const setListener = useSetRecoilState(listenerState);
  const setLocale = useSetRecoilState(localeState);
  const setOptions = useSetRecoilState(optionsState);
  const setProfileID = useSetRecoilState(profileIDState);
  const [loading, setLoading] = useRecoilState(loadingState);
  const {
    assessmentID,
    benchmarkID,
    locale,
    profileID,
    ...extraOptions
  } = options;

  useEffect(() => {
    setOptions(extraOptions);
  }, [options]);

  useEffect(() => {
    if(!locale) { return; }

    setLocale(locale);
  }, [locale]);

  useEffect(() => {
    const http = _http || new Http(slice(options, ["authKey", "host", "version"]));

    setHttp(http);
  }, [_http]);

  useEffect(() => {
    const i18n = _i18n || new I18n();

    setI18n(i18n);
  }, [_i18n]);

  useEffect(() => {
    const listener = _listener || new Listener();

    setListener(listener);
  }, [_listener]);

  useEffect(() => {
    setAssessmentID(assessmentID);
  }, [assessmentID]);

  useEffect(() => {
    setBenchmarkID(benchmarkID);
  }, [benchmarkID]);

  useEffect(() => {
    setProfileID(profileID);
  }, [profileID]);

  useEffect(() => {
    setLoading(false);
  }, []);

  if(loading) { return null; }

  return children;
}

State.defaultProps = {
  http: null,
  i18n: null,
  listener: null,
  options: {}
};
State.propTypes = {
  children: PropTypes.node.isRequired,
  http: PropTypes.object,
  i18n: PropTypes.object,
  listener: PropTypes.object,
  options: PropTypes.shape({
    assessmentID: PropTypes.string,
    benchmarkID: PropTypes.string,
    locale: PropTypes.string,
    profileID: PropTypes.string
  })
};

export default function Container(props) {
  return (
    <RecoilRoot>
      <State {...props} />
    </RecoilRoot>
  );
}
