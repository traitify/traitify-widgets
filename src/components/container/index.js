import {RecoilRoot} from "recoil";
import ErrorBoundary from "./error-boundary";
import State from "./state";

export default function Container(props) {
  return (
    <RecoilRoot>
      <ErrorBoundary>
        <State {...props} />
      </ErrorBoundary>
    </RecoilRoot>
  );
}
