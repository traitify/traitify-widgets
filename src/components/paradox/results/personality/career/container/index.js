import PropTypes from "prop-types";
import {useEffect} from "react";
import TraitifyPropTypes from "lib/helpers/prop-types";
import withTraitify from "lib/with-traitify";
import CareerFilter from "../filter";
import CareerModal from "../modal";
import CareerResults from "../list";
import style from "./style.scss";

function Careers(props) {
  const {ui, isReady} = props;

  useEffect(() => {
    ui.trigger("Careers.initialized", {props});
  }, []);
  useEffect(() => {
    ui.trigger("Careers.updated", {props});
  });

  return (
    isReady("results") && (
      <div className={style.container}>
        <CareerFilter {...props} />
        <CareerResults {...props} />
        <CareerModal {...props} />
      </div>
    )
  );
}
Careers.propTypes = {
  isReady: PropTypes.func.isRequired,
  ui: TraitifyPropTypes.ui.isRequired
};
export {Careers as Component};
export default withTraitify(Careers);
