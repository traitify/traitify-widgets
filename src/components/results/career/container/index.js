import useComponentEvents from "lib/hooks/use-component-events";
import useResults from "lib/hooks/use-results";
import style from "./style.scss";
import CareerFilter from "../filter";
import CareerList from "../list";
import CareerModal from "../modal";

export default function CareerContainer() {
  const results = useResults({surveyType: "personality"});

  useComponentEvents("Careers");

  if(!results) { return null; }

  return (
    <div className={style.container}>
      <CareerFilter />
      <CareerList />
      <CareerModal />
    </div>
  );
}
