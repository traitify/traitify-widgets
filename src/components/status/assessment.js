import {faCheck} from "@fortawesome/free-solid-svg-icons";
import PropTypes from "prop-types";
import {useSetRecoilState} from "recoil";
import Icon from "components/common/icon";
import get from "lib/common/object/get";
import useListener from "lib/hooks/use-listener";
import useOption from "lib/hooks/use-option";
import {activeState} from "lib/recoil";
import style from "./style.scss";

// TODO: Extract text to translate
const translations = {
  complete: "Complete",
  loading: "Loading", // Already translated
  status: {
    start: "Start Assessment"
  },
  survey: {
    cognitive_assessment: "Cognitive Assessment",
    external_assessment: "External Assessment",
    personality_assessment: "Personality Assessment"
  }
};

function Button({assessment}) {
  const listener = useListener();
  const options = useOption("status") || {};
  const redirect = get(options, "allowRedirect", true);
  const setActive = useSetRecoilState(activeState);

  if(assessment.completed) {
    return <button disabled={true} type="button">{translations.complete}</button>;
  }

  if(assessment.skipped) {
    return <button disabled={true} type="button">Skipped</button>;
  }

  if(assessment.link && redirect) {
    return <a href={assessment.link}>{translations.status.start}</a>;
  }

  if(assessment.loading) {
    return <button disabled={true} type="button">{translations.loading}</button>;
  }

  const start = () => {
    listener.trigger("Survey.start", {assessment});
    setActive({...assessment});
  };

  return <button onClick={start} type="button">{translations.status.start}</button>;
}

Button.propTypes = {
  assessment: PropTypes.shape({
    completed: PropTypes.bool,
    link: PropTypes.string,
    loading: PropTypes.bool,
    skipped: PropTypes.bool
  }).isRequired
};

function Assessment({assessment}) {
  const surveyName = assessment.surveyName || translations.survey[`${assessment.surveyType}_assessment`];

  return (
    <div className={[style.assessment, assessment.completed && style.inactive].filter(Boolean).join(" ")}>
      <div>
        {assessment.completed && <Icon className={style.icon} icon={faCheck} />}
        <span className={style.text}>{surveyName}</span>
      </div>
      <Button assessment={assessment} />
    </div>
  );
}

Assessment.propTypes = {
  assessment: PropTypes.shape({
    completed: PropTypes.bool,
    skipped: PropTypes.bool,
    surveyName: PropTypes.string,
    surveyType: PropTypes.string.isRequired
  }).isRequired
};

export default Assessment;
