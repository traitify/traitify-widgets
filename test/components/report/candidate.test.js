import Component from "components/report/candidate";
import ComponentHandler from "support/component-handler";
import {mockAssessment} from "support/container/http";
import useContainer from "support/hooks/use-container";
import assessment from "support/json/assessment/dimension-based.json";

jest.mock("components/results/personality/archetype/heading", () => (() => <div className="mock">Personality Archetype Heading</div>));
jest.mock("components/results/personality/archetype/tips", () => (() => <div className="mock">Personality Archetype Tips</div>));
jest.mock("components/results/personality/dimension/list", () => (() => <div className="mock">Personality Dimension List</div>));
jest.mock("components/results/personality/trait/list", () => (() => <div className="mock">Personality Traits</div>));
jest.mock("components/results/feedback", () => (() => <div className="mock">Feedback</div>));

describe("Report.Candidate", () => {
  let component;

  useContainer();

  it("renders component", async() => {
    mockAssessment(assessment, {id: assessment.id});
    component = await ComponentHandler.setup(Component);

    expect(component.tree).toMatchSnapshot();
  });
});
