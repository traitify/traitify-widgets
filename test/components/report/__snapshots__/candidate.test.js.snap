// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Report.Candidate renders component 1`] = `
<section
  className="container"
>
  <div
    className="mock"
  >
    Personality Archetype Heading
  </div>
  <div
    className="mock"
  >
    Personality Archetype Tips
  </div>
  <div
    className="mock"
  >
    Personality Dimension List
  </div>
  <div
    className="mock"
  >
    Personality Traits
  </div>
</section>
`;
